engine/common/rpc/convert/convert.go

```diff
@@ -813,6 +837,13 @@ func MessageToServiceEvent(m *entities.ServiceEvent) (*flow.ServiceEvent, error)
 			return nil, fmt.Errorf("failed to marshal to EpochCommit event: %w", err)
 		}
 		event = commit
+	case flow.ServiceEventVersionBeacon:
+		versionBeacon := new(flow.VersionBeacon)
+		err := json.Unmarshal(rawEvent, versionBeacon)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal to VersionBeacon event: %w", err)
```

Comment:
maybe dedup the Unmarshal calls.

e.g.,
```
var event := interface{}
switch m.Type {
case flow.ServiceEventSetup:
  event = new(flow.EpochSetup)
case flow.ServiceEventCommit:
  event = new(flow.EpochCommit)
case flow.ServiceEventVersionBeacon:
  event = new(flow.VersionBeacon)
default:
  return ...
}

err := json.Unmarshal(m.Payload, event)
if err != nil {
  return nil, fmt.Error("... %T event: %w", event, err)
}

return &flow.ServiceEvent{...}
```

Reply:
that's a slightly bigger unrelated refactor I put it here https://github.com/onflow/flow-go/pull/4220